3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/Adapter.scala

```diff
@@ -56,6 +59,64 @@ trait Adapter {
   // The encoding type to be used
   val EventEncType = "UTF-8"
 
+  // Datetime format we need to convert timestamps to
+  val JsonSchemaDateTimeFormat = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").withZone(DateTimeZone.UTC)
+
+  /**
+   * Returns an updated event JSON where
+   * all of the timestamp fields (tsFieldKey:_) have been
+   * changed to a valid JsonSchema date-time format
+   * and the "event":_type field has been removed
+   *
+   * @param json The event JSON which we need to
+   *        update values for
+   * @param eventOpt The event type as an Option[String]
+   *        which we are now going to remove from
+   *        the event JSON
+   * @param tsFieldKey the key name of the timestamp field
+   *                   which will be transformed
+   * @param toSeconds function which converts the numeric
+   *                  value of the key into a seconds past
+   *                  epoch format (e.g. x * 1000 for ms)
+   * @return the updated JSON with valid date-time
+   *         values in the tsFieldKey fields
+   */
+  private[registry] def cleanupJsonEventValues(json: JValue, eventOpt: Option[(String,String)], tsFieldKey: String, toSeconds: Long => Long): JValue = {
+
+    def toStringField(value: Long): JString = {
+      val dt: DateTime = new DateTime(toSeconds(value))
+      JString(JsonSchemaDateTimeFormat.print(dt))
+    }
+
+    val j1 = json transformField {
+      case (k, v) => {
+        if (k == tsFieldKey) {
+          v match {
+            case JInt(x) => {
+              try {
+                (k, toStringField(x.longValue()))
+              } catch {
+                case _: Throwable => (k, JInt(x))
+              }
+            }
+            case JString(x) => {
+              try {
+                (k, toStringField(x.toLong))
+              } catch {
+                case _: Throwable => (k, JString(x))
+              }
+            }
+          }
+        } else (k, v)
+      }
+    }
```

Comment:
Is it directly related to the [Java Throwable](http://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html)? 

Either way thanks - I'll change this, thinking about it if I saw some Java catching throwable I'd say the same thing.